sleap_nn/config/data_config.py

```diff
@@ -0,0 +1,166 @@
+import attrs
+from omegaconf import MISSING
+from typing import Optional, Tuple, List, Dict, Any
+
+
+"""Serializable configuration classes for specifying all data configuration parameters.
+
+These configuration classes are intended to specify all 
+the parameters required to initialize the data config.
+"""
+
+
+@attrs.define
+class PreprocessingConfig:
+    """Configuration of Preprocessing.
+
+    Attributes:
+        is_rgb: (bool) True if the image has 3 channels (RGB image). If input has only one channel when this is set to True, then the images from single-channel is replicated along the channel axis. If input has three channels and this is set to False, then we convert the image to grayscale (single-channel) image.
+        max_height: (int) Maximum height the image should be padded to. If not provided, the original image size will be retained. Default: None.
+        max_width: (int) Maximum width the image should be padded to. If not provided, the original image size will be retained. Default: None.
+        scale: (float or List[float]) Factor to resize the image dimensions by, specified as either a float scalar or as a 2-tuple of [scale_x, scale_y]. If a scalar is provided, both dimensions are resized by the same factor.
+        crop_hw: (Tuple[int]) Crop height and width of each instance (h, w) for centered-instance model. If None, this would be automatically computed based on the largest instance in the sio.Labels file.
+        min_crop_size: (int) Minimum crop size to be used if crop_hw is None.
+    """
+
+    is_rgb: bool = True
+    max_height: Optional[int] = None
```

Comment:
At top of file add:
```
from __futures__ import annotations
```
then we can use future python's support for type hinting.

```suggestion
    max_height: int | None = None
```

Reply:
attempted this change, but when running I get an error with omegaconf. It seems like omegaconf might not support this syntax
<img width="797" alt="Screenshot 2025-01-17 at 9 14 07 PM" src="https://github.com/user-attachments/assets/53326abd-47e2-4c01-bb64-facbca08f107" />